osbuild/util/mnt.py

```diff
@@ -44,14 +51,14 @@ class MountGuard(contextlib.AbstractContextManager):
     def __init__(self):
         self.mounts = []
 
-    def mount(self, source, target, bind=True, ro=False, rw=False, mode="0755"):
+    def mount(self, source, target, bind=True, permissions: Optional[MountPermissions] = None, mode="0755"):
         options = []
         if bind:
             options += ["bind"]
-        if ro:
-            options += ["ro"]
-        if rw:
-            options += ["rw"]
+        if permissions:
+            if permissions not in list(MountPermissions):
```

Comment:
So, theoretically because this is now typed it'd be impossible to pass anything in that *isn't* a variant of `MountPermissions` as it'd be caught by `mypy` but I'm also fine with keeping it.